src/internal/functions/Add-PuppetReferenceDocumentation.ps1

```diff
@@ -0,0 +1,42 @@
+function Add-PuppetReferenceDocumentation {
+    <#
+      .SYNOPSIS
+        Generate REFERENCE.md
+      .DESCRIPTION
+        Generate REFERENCE.md file for the Puppet module from the auto-generated types for each DSC
+        resource. This will *always* have the syntax but **may not** have the property documentation,
+        depending on whether or not those reference docs were discoverable for each DSC resource.
+      .PARAMETER PuppetModuleFolderPath
+        The path, relative or literal, to the Puppet module's root folder.
+      .EXAMPLE
+        Add-PuppetReferenceDocumentation -PuppetModuleFolderPath C:\output\testmodule
+        This command will generate `REFERENCE.md` file for the `testmodule` Puppet module.
+    #>
+    [CmdletBinding()]
+    param (
+      [Parameter(Mandatory=$true)]
+      $PuppetModuleFolderPath
+    )
+
+    begin {
+      $PuppetModuleFolderPath = Resolve-Path -Path $PuppetModuleFolderPath -ErrorAction Stop
+      $Command = [scriptblock]::Create("pdk bundle exec puppet strings generate --format markdown --out REFERENCE.md")
+    }
+    process {
+      Try {
+        $ErrorActionPreference = 'Stop'
+        Invoke-PdkCommand -Path $PuppetModuleFolderPath -Command $Command -SuccessFilterScript {
+          $_ -match "% documented"
+        }
+        # Verify REFERENCE.md file is generated
+        $ReferenceFile = Join-Path -Path $PuppetModuleFolderPath -ChildPath REFERENCE.md
+        if (!(Test-Path $ReferenceFile -PathType leaf))
+          {
+            Write-PSFMessage "REFERENCE.md file is not generated"
+          }
```

Comment:
Thinking about it a bit more, I think we could replace this with the below:

```suggestion
        $null = Resolve-Path $ReferenceFile
```

If the file can't be found, it will generate a terminating exception, which will trigger the `Catch` block and rethrow the error up the stack.